llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1246,32 +1244,16 @@ Value *HWAddressSanitizer::getHwasanThreadSlotPtr(IRBuilder<> &IRB, Type *Ty) {
   return nullptr;
 }
 
-Value *HWAddressSanitizer::getPC(IRBuilder<> &IRB) {
-  if (TargetTriple.getArch() == Triple::aarch64)
-    return readRegister(IRB, "pc");
-  return IRB.CreatePtrToInt(IRB.GetInsertBlock()->getParent(), IntptrTy);
-}
-
-Value *HWAddressSanitizer::getFP(IRBuilder<> &IRB) {
-  if (!CachedSP) {
-    // FIXME: use addressofreturnaddress (but implement it in aarch64 backend
-    // first).
-    Function *F = IRB.GetInsertBlock()->getParent();
-    Module *M = F->getParent();
-    auto *GetStackPointerFn = Intrinsic::getDeclaration(
-        M, Intrinsic::frameaddress,
-        IRB.getPtrTy(M->getDataLayout().getAllocaAddrSpace()));
-    CachedSP = IRB.CreatePtrToInt(
-        IRB.CreateCall(GetStackPointerFn, {Constant::getNullValue(Int32Ty)}),
-        IntptrTy);
-  }
+Value *HWAddressSanitizer::getCachedSP(IRBuilder<> &IRB) {
+  if (!CachedSP)
+    CachedSP = memtag::getSP(IRB);
```

Comment:
It's not NFC, it's FP (frameaddress) -> SP

Reply:
it's NFC, i just made the name consistent.

before, we had this

```
Value *SP = getFP(IRB);
```

Reply:
It should be getFP - frameaddress
getSP is sponentry/addressofreturnaddress or something like this

Reply:
I don't understand what we are talking about. If you say the naming is incorrect, yeah, maybe. But the change is obviously NFC, because all it does is
* rename getFP to getCachedSP
* pull out the `(!CachedSP)` into a function

Reply:
Sorry, when I commented I thought you are actually switching to SP, so I thought it's not NFC.